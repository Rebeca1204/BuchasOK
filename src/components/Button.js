import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Button = styled.button`
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: ${(props) => props.theme.colors.primary.main};
  border: 2px solid ${(props) => props.theme.colors.primary.main};
  color: ${(props) => props.theme.colors.primary.text};
  border-radius: 5px;
`;

Button.defaultProps = {
  type: "button",
  children: undefined,
};

Button.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
};
export default Button;
