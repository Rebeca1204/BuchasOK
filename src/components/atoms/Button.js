import PropTypes from "prop-types";
import styled from "styled-components";

export const ButtonColors = {
  default: "default",
  primary: "primary",
  danger: "danger",
};
export const ButtonVariants = {
  default: "default",
  outlined: "outlined",
  link: "link",
};

const getColor = ({ theme, color }) => {
  switch (color) {
    case ButtonColors.primary:
      return theme.colors.primary.main;
    case ButtonColors.danger:
      return theme.colors.danger.main;
    default:
      return "#e0e0e0";
  }
};
const getDarkColor = ({ theme, color }) => {
  switch (color) {
    case ButtonColors.primary:
      return theme.colors.primary.dark;
    case ButtonColors.danger:
      return theme.colors.danger.dark;
    default:
      return "#5a6268";
  }
};
const getOutlinedTextColor = (props) => {
  if (props.color === ButtonColors.default) {
    return "#212121";
  }
  return getColor(props);
};
const getLinkTextColor = (props) => {
  if (props.color === ButtonColors.default) {
    return "#757575";
  }
  return getColor(props);
};
const getTextColor = ({ theme, color }) => {
  switch (color) {
    case ButtonColors.primary:
      return theme.colors.primary.text;
    case ButtonColors.danger:
      return theme.colors.danger.text;
    default:
      return theme.colors.primary.text;
  }
};
const Button = styled.button`
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: ${getColor};
  border: 2px solid ${getColor};
  color: ${getTextColor};
  border-radius: 5px;

  &:disabled {
    opacity: 0.6;
    cursor: not-allowed;
  }

  &:hover:enabled {
    background-color: ${getDarkColor};
    border: ${getDarkColor};
  }
`;

const ButtonOutlined = styled(Button)`
  background-color: transparent;
  color: ${getOutlinedTextColor};

  &:hover:enabled {
    background-color: transparent;
    color: ${getDarkColor};
    border: 2px solid ${getDarkColor};
  }
`;

const ButtonLink = styled(Button)`
  background-color: transparent;
  border-color: transparent;
  color: ${getLinkTextColor};
  padding-left: 0;
  padding-right: 0;
  &:hover:enabled {
    background-color: transparent;
    border-color: transparent;
    color: ${getDarkColor};
  }
`;

const ButtonWrapper = (props) => {
  switch (props.variant) {
    case ButtonVariants.outlined:
      return <ButtonOutlined {...props} />;
    case ButtonVariants.link:
      return <ButtonLink {...props} />;
    default:
      return <Button {...props} />;
  }
};

ButtonWrapper.defaultProps = {
  type: "button",
  children: undefined,
  color: "default",
  variant: "default",
};

ButtonWrapper.propTypes = {
  type: PropTypes.string,
  children: PropTypes.node,
  color: PropTypes.oneOf(Object.values(ButtonColors)),
  variant: PropTypes.oneOf(Object.values(ButtonVariants)),
};
export default ButtonWrapper;
