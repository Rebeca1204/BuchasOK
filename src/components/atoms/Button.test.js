import React from "react";
import Button, { ButtonColors, ButtonVariants } from "./Button";
import { render, fireEvent } from "../../test-utils";

test("renders a text", () => {
  const { getByText } = render(<Button>Click</Button>);
  expect(getByText("Click")).toBeInTheDocument();
});

test("trigger event on click", () => {
  const handleClick = jest.fn();
  const { getByRole } = render(<Button onClick={handleClick} />);
  fireEvent.click(getByRole("button"));
  expect(handleClick).toBeCalled();
});

test.each(Object.values(ButtonColors).map((item) => [item]))(
  "renders with color %s",
  (color) => {
    const { asFragment } = render(<Button color={color} />);
    expect(asFragment()).toMatchSnapshot();
  },
);

test.each(Object.values(ButtonVariants).map((item) => [item]))(
  "renders variants %s",
  (variant) => {
    const { asFragment } = render(<Button variant={variant} />);
    expect(asFragment()).toMatchSnapshot();
  },
);

test("renders with color primary and variant outlined", () => {
  const { asFragment } = render(<Button variant="outlined" color="primary" />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders with color primary and variant link", () => {
  const { asFragment } = render(<Button variant="link" color="primary" />);
  expect(asFragment()).toMatchSnapshot();
});
