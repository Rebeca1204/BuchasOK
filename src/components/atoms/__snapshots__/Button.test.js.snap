// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders variants default 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #e0e0e0;
  border: 2px solid #e0e0e0;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #5a6268;
  border: #5a6268;
}

<button
    class="c0"
    color="default"
    type="button"
    variant="default"
  />
</DocumentFragment>
`;

exports[`renders variants link 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #e0e0e0;
  border: 2px solid #e0e0e0;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #5a6268;
  border: #5a6268;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #757575;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
  color: #5a6268;
}

<button
    class="c0 c1"
    color="default"
    type="button"
    variant="link"
  />
</DocumentFragment>
`;

exports[`renders variants outlined 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #e0e0e0;
  border: 2px solid #e0e0e0;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #5a6268;
  border: #5a6268;
}

.c1 {
  background-color: transparent;
  color: #212121;
}

.c1:hover:enabled {
  background-color: transparent;
  color: #5a6268;
  border: 2px solid #5a6268;
}

<button
    class="c0 c1"
    color="default"
    type="button"
    variant="outlined"
  />
</DocumentFragment>
`;

exports[`renders with color danger 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #6e4f4b;
  border: 2px solid #6e4f4b;
  color: #dfe0d8;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #b22a00;
  border: #b22a00;
}

<button
    class="c0"
    color="danger"
    type="button"
    variant="default"
  />
</DocumentFragment>
`;

exports[`renders with color default 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #e0e0e0;
  border: 2px solid #e0e0e0;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #5a6268;
  border: #5a6268;
}

<button
    class="c0"
    color="default"
    type="button"
    variant="default"
  />
</DocumentFragment>
`;

exports[`renders with color primary 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #90b19d;
  border: 2px solid #90b19d;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #324f49;
  border: #324f49;
}

<button
    class="c0"
    color="primary"
    type="button"
    variant="default"
  />
</DocumentFragment>
`;

exports[`renders with color primary and variant link 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #90b19d;
  border: 2px solid #90b19d;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #324f49;
  border: #324f49;
}

.c1 {
  background-color: transparent;
  border-color: transparent;
  color: #90b19d;
  padding-left: 0;
  padding-right: 0;
}

.c1:hover:enabled {
  background-color: transparent;
  border-color: transparent;
  color: #324f49;
}

<button
    class="c0 c1"
    color="primary"
    type="button"
    variant="link"
  />
</DocumentFragment>
`;

exports[`renders with color primary and variant outlined 1`] = `
<DocumentFragment>
  .c0 {
  font-size: 1rem;
  font-weight: 600;
  text-transform: uppercase;
  padding: 12px 36px;
  cursor: pointer;
  background-color: #90b19d;
  border: 2px solid #90b19d;
  color: #212121;
  border-radius: 5px;
}

.c0:disabled {
  opacity: 0.6;
  cursor: not-allowed;
}

.c0:hover:enabled {
  background-color: #324f49;
  border: #324f49;
}

.c1 {
  background-color: transparent;
  color: #90b19d;
}

.c1:hover:enabled {
  background-color: transparent;
  color: #324f49;
  border: 2px solid #324f49;
}

<button
    class="c0 c1"
    color="primary"
    type="button"
    variant="outlined"
  />
</DocumentFragment>
`;
